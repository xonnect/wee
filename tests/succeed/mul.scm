(define (foo x y)
  (* x y))

(displayln (foo 1 3))
(displayln (foo 15 3))
(displayln (* 21435 2141241))
(displayln (* 100 100))

(displayln (= (* 2 2) 4))
(displayln (= (* 0 5) 0))
(displayln (= (* 2 7) 14))
(displayln (= (* 1 -5) -5))
(displayln (= (* 2 -5) -10))
(displayln (= (* -7 5) -35))
(displayln (= (* -3 -7) 21))

(displayln (= (* 10000 10000) 100000000))
(displayln (= (* -10000 10000) -100000000))
(displayln (= (* 10000 -10000) -100000000))
(displayln (= (* -10000 -10000) 100000000))

(displayln (= (* 100000 10000) 1000000000))
(displayln (= (* 10000 100000) 1000000000))
(displayln (= (* 100000 100000) 10000000000))
