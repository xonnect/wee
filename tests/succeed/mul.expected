3
45
45897500835
10000
#t
#t
#t
#t
#t
#t
#t
#t
#t
#t
#t
#t
#t
#t
